### PR TITLE
Layout group

### PR DESCRIPTION
**Pull request by micahgodbolt**
Simple layout component to allow for vertical or horizontal layout of children elements. Nestable, and customizable. Currently uses flexbox for layout, but API is generic to consider grid in the future. 

image

### CODE REVIEW

File: packages/experiments/src/components/LayoutGroup/LayoutGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        break;
+      case 'center':
+        justifyValue = 'center';
+        break;
+      default:
+        justifyValue = 'flex-start'
+    };
+
+    const group = React.Children.map(children, (child: React.ReactChild, i: number) => {
+      const numberOfChildren = React.Children.count(children);
```
Comment (cschleiden) [inline packages/experiments/src/components/LayoutGroup/LayoutGroup.tsx]: move out of the loop?

File: packages/experiments/src/components/LayoutGroup/LayoutGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      return (
+        <div
+          className={
+            mergeStyles(
+              'ms-LayoutGroup-item',
+              direction === 'horizontal' && i < numberOfChildren - 1 && {
```
Comment (cschleiden) [inline packages/experiments/src/components/LayoutGroup/LayoutGroup.tsx]: can you extract this into a `const isLastChild = i === numberOfChildren -1`; variable?

File: packages/experiments/src/components/LayoutGroup/LayoutGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    switch (justify) {
+      case 'end':
+        justifyValue = 'flex-end';
+        break;
+      case 'center':
+        justifyValue = 'center';
+        break;
+      default:
+        justifyValue = 'flex-start'
+    };
```
Comment (cschleiden) [inline packages/experiments/src/components/LayoutGroup/LayoutGroup.tsx]: maybe extract this into a method?

Comment (micahgodbolt) [reply packages/experiments/src/components/LayoutGroup/LayoutGroup.tsx]: like this?

```ts
private _getJustify(prop: string | undefined): string {
    if (prop === 'end') {
      return 'flex-end';
    } else if (prop === 'center') {
      return 'center';
    } else {
      return 'flex-start';
    }
  }
```

**Pull request merged**